tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -215,33 +759,30 @@ function Get-AzureRmProfile
   }
   PROCESS
   {
+    # ListAvailable helps to display all profiles available from the gallery
     [switch]$ListAvailable = $PSBoundParameters.ListAvailable
     $ProfileMap = (Get-AzProfile @PSBoundParameters)
-    $AllProfiles = ($ProfileMap | Get-Member -MemberType NoteProperty).Name
     if ($ListAvailable.IsPresent)
     {
-        foreach ($Profile in $ProfileMap) 
-        { 
-            foreach ($Module in ($Profile | get-member -MemberType NoteProperty).Name)
-            {  
-                Write-Output "Profile: $Module"
-                Write-Output "----------------"
-                $Profile.$Module | fl
-            } 
-        }
+      Get-ProfilesAvailable $ProfileMap
     }
     else
     {
-      $result = @()
-      foreach ($key in $AllProfiles)
+      # Just display profiles installed on the machine
+      $IncompleteProfiles = @()
+      $profilesInstalled = Get-ProfilesInstalled -ProfileMap $ProfileMap ([REF]$IncompleteProfiles)
+      $Output = @()
+      foreach ($key in $profilesInstalled.Keys)
       {
-        if ((Get-AzureRmModule -Profile $key -Module $RollUpModule) -ne $null)
-        {
-          $result += $key
-        }
+        $Output += "Profile : $key"
```

Comment:
Profile names are dynamically retrieved. Powershell formatting requires the name to be specified to show as the header, which in this case is not possible. For example,
 $exp = @{Expression={$_.**$Profile**};Label="Profile";width=25},  @{Expression={$_.**$Module**};Label="Module";width=15}
Here, $Profile and $Module cannot be variables like this. This has to be a static property.